ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSjoystick.cpp

```diff
@@ -32,108 +32,103 @@ namespace enigma_user
 {
-	JOYINFO joyinfo;
-	UINT wNumDevs;
-	BOOL bDevAttached;
-
-    if((wNumDevs = joyGetNumDevs()) == 0)
-        return false;
-
-    if (joySetCapture(enigma::hWnd, JOYSTICKID1 + id, 0, FALSE))
-    {
-        MessageBeep(MB_ICONEXCLAMATION);
-        MessageBox(enigma::hWnd, "Couldn't capture the joystick.", NULL,
-            MB_OK | MB_ICONEXCLAMATION);
-        return false;
-    }
-
-	bDevAttached = joyGetPos(JOYSTICKID1 + id, &joyinfo) != JOYERR_UNPLUGGED;
-    if (!bDevAttached)
-        return false;
-
-	return true;
+  JOYINFO joyinfo;
+  UINT wNumDevs;
+  BOOL bDevAttached;
+
+  if((wNumDevs = joyGetNumDevs()) == 0)
+    return false;
+
+  if (joySetCapture(enigma::hWnd, JOYSTICKID1 + id, 0, FALSE))
+  {
+    MessageBeep(MB_ICONEXCLAMATION);
+    MessageBox(enigma::hWnd, "Couldn't capture the joystick.", NULL,
+        MB_OK | MB_ICONEXCLAMATION);
+    return false;
+  }
+
+  bDevAttached = joyGetPos(JOYSTICKID1 + id, &joyinfo) != JOYERR_UNPLUGGED;
+  if (!bDevAttached)
+    return false;
+
+  return true;
 }
 
 double joystick_axis(int id, int axisnum) {
-	JOYINFOEX joyinfo;
-	JOYCAPS joycaps;
-    joyGetDevCaps(JOYSTICKID1 + id, &joycaps, sizeof(joycaps));
-	switch (axisnum) {
-		case 1:
-			joyinfo.dwFlags = JOY_RETURNX | JOY_RETURNCENTERED;
-			joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
-			return (joyinfo.dwXpos - (joycaps.wXmax + joycaps.wXmin) / 2.f) / ((joycaps.wXmax - joycaps.wXmin)/ 2.f);
-			break;
-		case 2:
-			joyinfo.dwFlags = JOY_RETURNY | JOY_RETURNCENTERED;
-			joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
-			return (joyinfo.dwYpos - (joycaps.wYmax + joycaps.wYmin) / 2.f) / ((joycaps.wYmax - joycaps.wYmin)/ 2.f);
-			break;
-		case 3:
-			joyinfo.dwFlags = JOY_RETURNZ | JOY_RETURNCENTERED;
-			joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
-			return (joyinfo.dwZpos - (joycaps.wZmax + joycaps.wZmin) / 2.f) / ((joycaps.wZmax - joycaps.wZmin)/ 2.f);
-			break;
-		case 4:
-			joyinfo.dwFlags = JOY_RETURNR | JOY_RETURNCENTERED;
-			joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
-			return (joyinfo.dwRpos - (joycaps.wRmax + joycaps.wRmin) / 2.f) / ((joycaps.wRmax - joycaps.wRmin)/ 2.f);
-			break;
-		case 5:
-			joyinfo.dwFlags = JOY_RETURNU | JOY_RETURNCENTERED;
-			joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
-			return (joyinfo.dwUpos - (joycaps.wUmax + joycaps.wUmin) / 2.f) / ((joycaps.wUmax - joycaps.wUmin)/ 2.f);
-			break;
-		case 6:
-			joyinfo.dwFlags = JOY_RETURNV | JOY_RETURNCENTERED;
-			joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
-			return (joyinfo.dwVpos - (joycaps.wVmax + joycaps.wVmin) / 2.f) / ((joycaps.wVmax - joycaps.wVmin)/ 2.f);
-			break;
-		default:
-			return 0;
-	}
+  JOYINFOEX joyinfo;
+  JOYCAPS joycaps;
+  joyGetDevCaps(JOYSTICKID1 + id, &joycaps, sizeof(joycaps));
+  switch (axisnum) {
+    case 1:
+      joyinfo.dwFlags = JOY_RETURNX | JOY_RETURNCENTERED;
+      joyGetPosEx(JOYSTICKID1 + id, &joyinfo);
+      return (joyinfo.dwXpos - (joycaps.wXmax + joycaps.wXmin) / 2.f) / ((joycaps.wXmax - joycaps.wXmin)/ 2.f);
```

Comment:
Please extract a `inline float joystick_grade(float position, float min, float max)`.
The body should just be `(pos - (max + min) / 2.f) / ((max - min)/ 2.f);`.
Am I correct in saying that min and max can be declared `constexpr`? Try it.